src/NzbDrone.Core/Download/Clients/Sabnzbd/Sabnzbd.cs

```diff
@@ -195,13 +196,46 @@ public override IEnumerable<DownloadClientItem> GetItems()
 
         public override void RemoveItem(string downloadId, bool deleteData)
         {
-            if (GetQueue().Any(v => v.DownloadId == downloadId))
+            var historyItem = GetHistory().SingleOrDefault(v => v.DownloadId == downloadId);
+
+            if (historyItem == null)
             {
                 _proxy.RemoveFrom("queue", downloadId, deleteData, Settings);
             }
             else
             {
                 _proxy.RemoveFrom("history", downloadId, deleteData, Settings);
+
+                // Completed items in SAB's history do not remove the files from the file system when deleted, even if instructed to.
+                // If the output path is valid delete the file(s), otherwise warn that they cannot be deleted.
+                if (deleteData && historyItem.Status == DownloadItemStatus.Completed)
+                {
+                    if (ValidatePath(historyItem))
+                    {
+                        var outputPath = historyItem.OutputPath;
+
+                        try
+                        {
+                            if (_diskProvider.FolderExists(outputPath.FullPath))
+                            {
+                                _diskProvider.DeleteFolder(outputPath.FullPath.ToString(), true);
+                            }
+                            else if (_diskProvider.FileExists(outputPath.FullPath))
+                            {
+                                _diskProvider.DeleteFile(outputPath.FullPath.ToString());
+                            }
+                        }
+                        catch (Exception e)
+                        {
+                            _logger.Error("Unable to delete output path: '{0}'. Delete file(s) manually", outputPath.FullPath);
+                            _logger.Trace(e, "Unable to delete output path");
```

Comment:
this was to get rid of the error for e being defined but not used - is this an alright way to do it here or is there a preferred way?

Reply:
Just remove e and call it as `catch (Exception)`